### DISCUSSION TITLE
Concept

### DISCUSSION THREAD

**Discussion by aljawaid** [aljawaid/TemplateManager#1] [Ideas]
Project task description templates already exist in the core with a database table (predefined contents).

Extend this to task comments as templates also, maybe in the same database table if possible.  Ifso, Database table would need to add a `type` (description or comment) column and a `label` (for future short onscreen labels for he user) column.

The comment templates would be shown in the comment editor.

The task description templates already show in the core so its just the comment templates functionality needed

**Comment by aljawaid** [comment]
@creecros your insight would be greatly appreciated

**Reply by creecros** [reply]
not something i have ever pondered, or played with. I would need to study.

**Reply by aljawaid** [reply]
I have hardcode templates for comments for 3 projects.  Im trying to make them like the taskdescriptions predefined content.

I think a seperate database table would probably be better... I just thought it may be easer to add 2 new columns in the existing table and stick to that as its a simple table and then tap into the same controllers or extend them etc

**Reply by aljawaid** [reply]
@creecros by the way I set up sponsoring on my GitHub for all my repositories. Not expecting anything, but never say never.

Apparently, we can set up to 4 usernames so if you setup the sponsor profile on your account let me know, then I will add your username to the relevant repositories.

**Reply by creecros** [reply]
i enabled mine, but i never set it up. when they start asking for me to link my bank accounts, it becomes a joint effort with my wife, cause im clueless.

**Comment by aljawaid** [comment]
Will be using the new datatbase table approach

**Comment by aljawaid** [comment]
@creecros I've managed to get 90% of this plugin done but I am struggling with just one function - could you please help? as I am getting no errors, just a spinning button.

I am deleting a predefined task description. The delete button shows a modal fine, but when I press delete inside the modal, it forevers just spins.
https://github.com/aljawaid/TemplateManager/blob/58dc01ad636d43703a7c1ac682f8909addd37f79/Template/predefined_task_description/remove.php#L29

I copied the controller from the core and all my other templates/functions work. I'd appreciate it and advise if you could spot a mistake.

**Comment by creecros** [comment]
my invite expired.

**Comment by creecros** [comment]
as far as your modal, you need to define the plugin.
```
    <?= $this->modal->confirmButtons(
        'TaskDescriptionTemplateController',
        'remove',
        array('plugin' => 'templateManager', 'project_id' => $project['id'], 'id' => $template['id']),
        t('Delete')
    ) ?>
```
that said...
this plugin does not even function for me. schema is wrong(sqlite), and only the task description works for me.

## Configuration
- Application version: v1.2.26
- PHP version: 8.1.14
- PHP SAPI: fpm-fcgi
- HTTP Client: cURL
- OS version: Linux 5.10.102.1-microsoft-standard-WSL2
- Database driver: sqlite
- Database version: 3.40.1
- Browser: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36

**Reply by aljawaid** [reply]
oh ok, I followed the examples.... the mysql works fine for me... i dont have sqlite on my live setup.... any suggestions what the format should be?

**Reply by aljawaid** [reply]
> as far as your modal, you need to define the plugin.
> 
> ```
>     <?= $this->modal->confirmButtons(
>         'TaskDescriptionTemplateController',
>         'remove',
>         array('plugin' => 'templateManager', 'project_id' => $project['id'], 'id' => $template['id']),
>         t('Delete')
>     ) ?>
> ```
- THANKS! I spent 3 days trying all sorts without realising...

**Comment by creecros** [comment]
having a tough time figuring out whats wrong, still. might need some time on this one.

**Reply by aljawaid** [reply]
no no you fixed it... it works after declaring the plugin name

**Reply by aljawaid** [reply]
or did you mean the sqlite is still not working after your fixes?

**Reply by aljawaid** [reply]
I've developed using php7.4 if it helps... but as of now, I am working fine with no errors or issues in this plugin since your commit but i use mysql. (im working on the `notes` feature right now, frontend + bit of backend)

**Reply by creecros** [reply]
i mean it does not work for me. i get "Unable to create this template" for everything but the task description.

**Reply by aljawaid** [reply]
> i mean it does not work for me. i get "Unable to create this template" for everything but the task description.
- surely thats an sqlite issue?

**Reply by creecros** [reply]
i dont know, currently. i can test in mysql as well in a min. but i do see other things that are peaking my interest. i.e., i dont see anyway to save all the values in the model. there is a lot of code here, so i need time.

**Comment by creecros** [comment]
no, doesnt work in MYSQL either

## Configuration
- Application version: v1.2.26
- PHP version: 8.1.14
- PHP SAPI: fpm-fcgi
- HTTP Client: cURL
- OS version: Linux 5.10.102.1-microsoft-standard-WSL2
- Database driver: mysql
- Database version: 8.0.31
- Browser: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36

**Comment by creecros** [comment]
oh i see why

**Reply by creecros** [reply]
i thinks it becuase you have note not null, but you aren't putting in anything for 'note'

**Reply by creecros** [reply]
yes, that's what it was, now let me test again in SQLITE.

**Reply by creecros** [reply]
yup works now.

**Reply by creecros** [reply]
ill let you take care of the rest, but you have to persist the value if it is defined as 'NOT NULL' in the schema. hence the reason nothing worked for me, since you did not persist 'note' yet defined it as 'NOT NULL'

i only did the comment on save, to test and make sure everything worked, but there are quite a few places you will need to fix.

**Reply by creecros** [reply]
also your POSTGRES is incorrect schema as well.

**Reply by creecros** [reply]
and to make that more clear, by putting 'NOT NULL' it becomes a requirement, which your modal only shows title as required, whereas your schema shows title,description and note as required.

![image](https://user-images.githubusercontent.com/26339368/219721663-09471852-9106-4edd-b86d-8f97139407ae.png)

**Reply by aljawaid** [reply]
ok, I will look into it all again... I only put NOT NULL as that was in the example, so I thought it had to be included... Even the examples in Kanboard docs are totally shit!
